src/pages/create.tsx

```diff
@@ -0,0 +1,183 @@
+const CreateProgramPage = () => {
+  const [stage, setStage] = useState(0);
+  const [programName, setProgramName] = useState("");
+  const [programLogo, setProgramLogo] = useState("");
+  const [programIdentifier, setProgramIdentifier] = useState("");
+  //const [programIsPublic, setProgramIsPublic] = useState(true); // TODO: add checkbox on form for setting program public or not
+  const [createProgram] = useCreateProgramMutation();
+  const [displayError, setError] = useState(""); // TODO: Proper error box
+  const [loading, setLoading] = useState(false);
+
+  const stepOne = () => {
+    return (
+      <div className="flex flex-col w-full px-4 py-12 md:w-152">
+        <Text h1 b className="whitespace no-wrap">
+          Create your mentorship!
+        </Text>
+        <div className="h-6" />
+
+        <Text className="text-secondary">
+          What is the name of your mentorship program?
+        </Text>
+        <div className="h-6" />
+
+        <div className="w-full">
+          <Input
+            name="Program Name"
+            placeholder="e.g. Nova Mentorship"
+            value={programName}
+            className="w-full"
+            onChange={(e) => {
+              setProgramName(e.target.value);
+            }}
+          />
+        </div>
+        <div className="h-3" />
+
+        <Button
+          disabled={programName.length == 0}
+          onClick={() => {
+            if (programName) {
+              setStage((prev) => prev + 1);
+              setError("");
+            } else setError("Please enter a program name.");
+          }}
+        >
+          Next
+        </Button>
+        <div className="h-6" />
+
+        <Text className="text-error">{displayError}</Text>
+      </div>
+    );
+  };
+
+  const stepTwo = () => {
+    return (
+      <div className="flex flex-col w-full px-4 py-12 md:w-120">
+        <Text h1 b>
+          Program Details
+        </Text>
+        <Text>
+          Let's set up some basic information! You can also do these later.
+        </Text>
+        <div className="h-6" />
+        <div>
+          <Text b>Logo</Text>
+          <div className="h-1" />
+          <Button variant="inverted" size="small">
+            Choose File
+          </Button>
+          <input // TODO: display preview of logo
+            type="file"
+            name="Program Logo"
+            value={programLogo}
+            className="flex-1"
+            onChange={(e) => {
+              setProgramLogo(e.target.value);
+            }}
+          />
+        </div>
+
+        <div className="h-6" />
+        <Text b>Identifier</Text>
+        <div className="inline">
+          <Text b2>www.mentorcenter.us/</Text>
+          <Input
+            title="Identifier"
+            name="Program Identifier"
+            placeholder="nova-mentorship"
+            value={programIdentifier}
+            className="flex-1"
+            onChange={(e) => {
+              setProgramIdentifier(e.target.value);
```

Comment:
Are we allowing entries like " " for the name of mentorships as well as the identifier? Probably need to get backend to run these checks because it didn't return an error for me although I'm certain we wouldn't be able to have spaces in the identifier

Reply:
checking for whitespace now and non-url friendly chars (space, slash) in the identifier as well